pkg/clients/aws.go

```diff
@@ -92,6 +98,45 @@ func LoadConfig(data []byte, profile, region string) (*aws.Config, error) {
 	return &config, err
 }
 
+// LoadSAConfig assumes an IAM role configured via a ServiceAccount.
+// https://docs.aws.amazon.com/eks/latest/userguide/iam-roles-for-service-accounts.html
+func LoadSAConfig(ctx context.Context, region string) (*aws.Config, error) {
+	cfg, err := external.LoadDefaultAWSConfig()
+	if err != nil {
+		return nil, err
+	}
+	cfg.Region = region
+	svc := sts.New(cfg)
+
+	b, err := ioutil.ReadFile(os.Getenv("AWS_WEB_IDENTITY_TOKEN_FILE"))
```

Comment:
It might be worth returning a distinct error if this `os.Getenv` call returns the empty string. My guess is that if this env var is not set we'll return an error like `cannot read file: ""`. Though that said, the logs will include a stack trace which may be sufficient.